modules/mergeImageWithTelemetry/mergeImageWithTelemetry.py

```diff
@@ -0,0 +1,84 @@
+from modules.mergeImageWithTelemetry.mergedData import MergedData
+from modules.timestamp.timestamp import Timestamp
+
+import datetime
+import numpy.typing as npt
+import typing
+
+class MergeImageWithTelemetry:
+    """
+    Merges image pipeline with telemetry pipeline
+    ...
+
+    Attributes
+    ----------
+    curData : Timestamp 
+        the current telemetry data 
```

Comment:
By "current" do you mean oldest in the list?

Reply:
There seems to be a lot of unnecessary code dealing with `curData` (whether it contains valid data at the start, updating it when it's matched, etc.). Since there are already checks for `len(telemetryData) == 0`, why not use `telemetryData[0]` directly?